Dockerfile

```diff
@@ -67,6 +72,7 @@ RUN apt-get update && apt-get install -y \
 	xfsprogs \
 	libzfs-dev \
 	tar \
+	linux-image-`uname -r` \
```

Comment:
@estesp thanks for the clarification.  I think my expectation was based on the old devenvironment documentation (https://docs.docker.com/v1.5/contributing/devenvironment/).  I noticed that I can't find that file in current source tree (although there are still some references to it in a few comments).  Anyway your explanation about dynamic linking makes sense as to why the generating the Docker binary can no longer be isolated from the environment where the binary will be used.

I noticed that the contrib directory has a note that says: 

> The `contrib` directory contains scripts, images, and other helpful things which are not part of the core docker distribution. Please note that they could be out of date, since they do not receive the same attention as the rest of the repository.

Does this mean that all of the development work for Docker Linux is done using Ubuntu 14.04 host systems (as referenced in the main `Dockerfile`), and then later ported to other systems later?